frontend/tests/unit/composables/useRequestFee.spec.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { flushPromises } from '@vue/test-utils';
+import type { Ref } from 'vue';
+import { ref } from 'vue';
+
+import { useRequestFee } from '@/composables/useRequestFee';
+import * as requestManagerService from '@/services/transactions/request-manager';
+import type { IEthereumProvider } from '@/services/web3-provider';
+import { TokenAmount } from '@/types/token-amount';
+import { UInt256 } from '@/types/uint-256';
+import {
+  generateTokenAmountData,
+  generateUInt256Data,
+  getRandomEthereumAddress,
+} from '~/utils/data_generators';
+import { MockedMetaMaskProvider } from '~/utils/mocks/ethereum-provider';
+
+vi.mock('@/services/transactions/request-manager');
+
+describe('useRequestFee', () => {
+  beforeEach(() => {
+    requestManagerService!.getRequestFee = vi
+      .fn()
+      .mockResolvedValue(new UInt256(generateUInt256Data()));
+  });
+
+  describe('available', () => {
+    it('should be false when there is no provider', () => {
+      const requestManagerAddress = ref(getRandomEthereumAddress());
+      const requestAmount = ref(new TokenAmount(generateTokenAmountData())) as Ref<TokenAmount>;
```

Comment:
Ok makes sense. So I will only define the values locally in the function if the values are actually tested.

Reply:
Yeah kinda. It won't be always possible and sometimes it makes sense to have exceptions because it improves the readability or so. But I think in general it is a good approach to keep test cases easy to read and understandable.

Reply:
👌🏾